04.ls/ls.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+COLUMN = 3
+SPACE = 2
+
+def filter_names(names)
+  names.reject { |name| name.start_with?('.') }.sort
+end
+
+def reposition(names)
+  row = (names.size.to_f / COLUMN).ceil
+  total_blank = names.size % COLUMN
+
+  sliced_names_front = names.slice(0...-total_blank)
+  sliced_names_back = names.slice(-total_blank..-1)
+
+  sliced_names_by_row = if row == 2
+                          sliced_names_front.each_slice(row).to_a.push(sliced_names_back)
+                        else
+                          names.each_slice(row)
+                        end
+
+  sliced_names_by_row.map do |names_by_row|
+    names_by_row.values_at(0...row)
+  end.transpose
+end
+
+def find_max_str_size(names)
+  str_sizes = []
+
+  names.each do |strings|
+    strings.each_with_index do |str, col|
+      str_sizes[col] ||= []
+      str_sizes[col] << str.size unless str.nil?
```

Comment:
```suggestion
      str_sizes[col] << str.to_s.size
```

好みの問題ですが条件分岐を消した方がスッキリしそうです
（49行目も同様）